apps/backend/src/modules/quiz-answers/quiz-answer.service.ts

```diff
@@ -0,0 +1,179 @@
+import { ErrorMessage } from "~/libs/enums/enums.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+import { type Service } from "~/libs/types/types.js";
+
+import {
+	type CategoryService,
+	type QuizScoreResponseDto,
+} from "../categories/categories.js";
+import { type QuizQuestionService } from "../quiz-questions/quiz-questions.js";
+import { INITIAL_STATISTIC_VALUE } from "./libs/constants/constants.js";
+import { QuizError } from "./libs/exceptions/exceptions.js";
+import {
+	type CategoryStatistic,
+	type QuizAnswerDto,
+	type QuizAnswerRequestDto,
+	type QuizAnswersResponseDto,
+	type QuizUserAnswerDto,
+	type UserAnswersRequestDto,
+} from "./libs/types/types.js";
+import { QuizAnswerEntity } from "./quiz-answer.entity.js";
+import { type QuizAnswerRepository } from "./quiz-answer.repository.js";
+
+class QuizAnswerService implements Service {
+	private categoryService: CategoryService;
+
+	private quizAnswerRepository: QuizAnswerRepository;
+
+	private quizQuestionService: QuizQuestionService;
+
+	public constructor(
+		quizAnswerRepository: QuizAnswerRepository,
+		categoryService: CategoryService,
+		quizQuestionService: QuizQuestionService,
+	) {
+		this.quizAnswerRepository = quizAnswerRepository;
+		this.categoryService = categoryService;
+		this.quizQuestionService = quizQuestionService;
+	}
+
+	public convertAnswerEntityToDto(
+		answerEntity: QuizAnswerEntity,
+	): QuizAnswerDto {
+		const answer = answerEntity.toObject();
+		const userAnswers: QuizUserAnswerDto[] = answer.userAnswers.map(
+			(userAnswer: QuizAnswerEntity) => {
+				return userAnswer.toObject() as QuizUserAnswerDto;
+			},
+		);
+
+		return {
+			...answer,
+			userAnswers,
+		};
+	}
+
+	public async create(payload: QuizAnswerRequestDto): Promise<QuizAnswerDto> {
+		const answerEntity = await this.quizAnswerRepository.create(
+			QuizAnswerEntity.initializeNew(payload),
+		);
+
+		return this.convertAnswerEntityToDto(answerEntity);
+	}
+
+	public async createScores({
+		answerIds,
+		userId,
+	}: UserAnswersRequestDto): Promise<QuizScoreResponseDto> {
+		const categoryStatistics = new Map<number, CategoryStatistic>();
+		const scores = [];
+
+		const categorizedAnswers = await Promise.all(
+			answerIds.map((answerId) => {
+				return this.quizAnswerRepository.getCategorizedAnswer(answerId);
+			}),
+		);
+
+		for (const answer of categorizedAnswers) {
+			const { categoryId, value } = answer;
+			const statistic = categoryStatistics.get(categoryId) || {
+				accumulatedSum: INITIAL_STATISTIC_VALUE,
+				categoryCount: INITIAL_STATISTIC_VALUE,
+			};
+			statistic.accumulatedSum += value;
+			statistic.categoryCount += 1;
+			categoryStatistics.set(categoryId, statistic);
+		}
+
+		for (const [categoryId, stats] of categoryStatistics.entries()) {
```

Comment:
Let's replace this for with 
```const scores = categoryStatistics.entries().map(([categoryId, stats]) => {//return userScore})```
It will be more "TS-way" to solve this mapping task